src/arith/rewrite_simplify.cc

```diff
@@ -1698,6 +1700,11 @@ PrimExpr RewriteSimplifier::Impl::ApplyRewriteRules(EQ ret) {
     TVM_TRY_REWRITE(c1 - x == c2, x == c1 - c2);
     TVM_TRY_REWRITE(x + c1 == c2, x == c2 - c1);
     TVM_TRY_RECURSIVE_REWRITE(x * y == 0, x == 0 || y == 0);
+    TVM_TRY_REWRITE(x == x, ctrue);
+  } else {
```

Comment:
Hi, I have one question on that generally `NaN == NaN` evaluates to False. I think it would be great to leave some explaination on the case. For example, could we just define it as some "undefined" behaviour?

Reply:
Good point, and I'll add a comment on it.  In general, the numeric simplifications assume that the values are neither `NaN` nor `Inf`, and that expressions do not rely on integer overflow for correctness.  That is, integer types and floating-point types are simplified as if they are mathematical integer/real numbers, rather than IEEE floats.